rele/client.py

```diff
@@ -58,17 +58,22 @@ def create_subscription(self, subscription, topic):
         :param topic: str Topic name to subscribe
         """
         subscription_path = self._client.subscription_path(
-            self._gc_project_id, subscription
+            self._gc_project_id, subscription.name
         )
-        topic_path = self._client.topic_path(self._gc_project_id, topic)
+        topic_path = self._client.topic_path(self._gc_project_id, subscription.topic)
 
         with suppress(exceptions.AlreadyExists):
             try:
-                self._client.create_subscription(
-                    name=subscription_path,
-                    topic=topic_path,
-                    ack_deadline_seconds=self._ack_deadline,
-                )
+                request = {
+                    "name": subscription_path,
+                    "topic": topic_path,
+                    "ack_deadline_seconds": self._ack_deadline,
+                }
+
+                if subscription.filter_expression:
+                    request["filter"] = subscription.filter_expression
+
+                self._client.create_subscription(request=request)
```

Comment:
is this 2.0 syntax?

Reply:
This is an alternative it always had apparently, I can't find if it's something new. You can choose both ways of invoking the method.